specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2021-07-01/managedClusters.json

```diff
@@ -4183,6 +4186,28 @@
       "title": "The max price (in US Dollars) you are willing to pay for spot instances. Possible values are any decimal value greater than zero or -1 which indicates default price to be up-to on-demand.",
       "description": "Possible values are any decimal value greater than zero or -1 which indicates the willingness to pay any on-demand price. For more details on spot pricing, see [spot VMs pricing](https://docs.microsoft.com/azure/virtual-machines/spot-vms#pricing)"
     },
+    "ScaleDownMode": {
```

Comment:
probably too late (and sorta minor) but: `ScaleDownMode` sorta makes me think it'd only impact scale down and not grow. The fact that this impacts grows behavior too sorta makes it feel more like a `ScalingMode` or `ResizeMode` or `ScaleMode` or something like that (i.e. it's not specifically _down_).

Reply:
For my learning: what's the right time/place to notice these sorts and things and ask them in the feature impl process?

Reply:
yeah good point. I've always had that in the back of my mind. It _might_ be possible to change it. PRD review would be the best time.

The mode manually concerns scale down so i guess that's where the name was sort of derived from - the behaviour is:
- If mode is delete, scaling up and down works by adding and removing instances.
- If mode is deallocate, check if there are any pre-existing deallocated instances. If I can satisfy the scale up with that amount, then start those. If not, then start what I can and also create new instances to satisfy the count. Scale down just deallocates VMs.

Reply:
Did we make this name up or did we pick it from VMSS?

Reply:
made it up. it's not a property on other API afaik.